lib/src/genius.dart

```diff
@@ -403,30 +408,42 @@ class Genius {
   ///}
   /// ```
   /// {@end-tool}
-  Future<Song?> searchSongByLyricsSnippet(
-      {required String lyricsSnippet, bool getFullInfo = true}) async {
+  Future<List<Song>?> searchSongsByLyricsSnippet({
+    required String lyricsSnippet,
+    bool getFullInfo = false,
+    int perPage = 10,
+    int page = 1,
+  }) async {
     Map<String, dynamic>? response = await _httpClient.makeRequest(
-        url: searchByLyricsSnippetRoute,
-        query: {'q': lyricsSnippet},
+        url: searchSongsByLyricsSnippetRoute,
+        query: {
+          'q': lyricsSnippet,
+          'page': page.toString(),
+          'per_page': perPage.toString()
+        },
         headers: false);
 
     if (response == null) {
       return _verbosePrint('error making the request');
     } else {
-      List<dynamic> sections = response['sections'];
+      List<dynamic> hits = response['sections'][0]['hits'];
 
-      Map<String, dynamic>? lyricSection =
-          firstWhereOrNull(sections, (element) => element['type'] == 'lyric');
+      List<Song> songs = [];
```

Comment:
dart recomend's create list like this `var songs = <Songs>[]` , [here](https://dart.dev/guides/libraries/library-tour#lists)

Reply:
I couldn't find anywhere in the link that recommends it. I believe that  `var songs = <Songs>[]` is a choice as both syntaxes achieve the same thing.